### DISCUSSION TITLE
Don't automatically untoggle when I switch files

### DISCUSSION THREAD

**Discussion by corysimmons** [moalamri/vscode-inline-fold#101] [Ideas]
If I use `Inline Fold: Toggle` in VSCode Command Palette, it works fine (i.e. unfolds) on the current file. If I open another file, it goes back to folding everything...

As a feature request, I think the behavior of that `Toggle` action should be permanent. In lieu of that, introduce two new actions like:

- `Inline Fold: Disable until I turn it back on`
- `Inline Fold: Enable until I turn it off

**Comment by moalamri** [comment]
Hey @corysimmons 
There's an option `inlineFold.autoFold` that does the exact thing, it doesn't have a command, do you think such option should have a registered command?

**Reply by corysimmons** [reply]
Thanks for the quick reply. I tried that out but it seems to act kind of buggy.

If I disable that setting and toggle off Inline Fold then open a file it sometimes is disabled, and for other files it seems enabled.

It seems to persist some sort of previous toggle state of each file.

In my ideal scenario there would just be an action where I could basically disable and enable the plugin _completely_. An action like `Inline Fold: Turn off` that changes to `Inline Fold: Turn on` (and vice versa).

My use case is if I'm touching Tailwind classes, I'm probably doing _a lot_ of that, so I'd like to hop around files and see all my classes expanded. Versus when I'm working on business logic I want to hide all my class names and just work on JS. Having some VSCode action I can bind a hotkey too would be amazing.

**Reply by moalamri** [reply]
Yes it is persistent per file and can be reset by the command ‘Inline Fold: Clear session cache’. It is because most of the users wanted changes to be per file and not per workspace. So I added that command to reset all saved states (per file) and adapt to the new option value. It seems to be a bit annoying if the whole workspace changes to a single state so by adding those options and commands things would be more flexible.
In your scenario you would have to manually change the option to ‘autoFold: false’ and run the reset command.
Perhaps I could add a command to change the default state and do a complete reset 🤔

**Reply by corysimmons** [reply]
Yeah, I see the issue. That will work, but doesn't it two-step the process every time I want to disable/enable Inline Fold?

I think an additional VSCode action like `Inline Fold: Turn off` | `Inline Fold: Turn on` would be non-breaking + ideal.

**Reply by moalamri** [reply]
Yes, a single command to fire both. The command name could be something like `Inline Fold: Toggle & Clear session`
As far as I know, the command name is static and is defined in the extension's `package.json`.
So `on` and `off` are going to be separated commands, I see that a single toggler is better, do you agree?

**Reply by corysimmons** [reply]
Yeah that'd work.

Basically anything to globally toggle the plugin on/off that I can bind to a hotkey works for me! :D
